### ISSUE TITLE
edge contrast penalty clarification

### ISSUE THREAD

**Issue by eyaler** [arthurmeyer/Saliency_Detection_Convolutional_Autoencoder#1]
could you kindly add some references or explanation for the edge contrast penalty? thanks!

**Comment by arthurmeyer** [comment]
Hey,
The idea is basically that cross entropy loss only consider pixel-wise error. To improve performance I also used the spatial information about pixels contained in the ground truth. This is what is called edge contrast penalty and is included to the loss during training. Specifically for each pixel on the edge of the salient object of the ground truth, I consider its four neighbors (top, bottom, left and right). Afterward the network try to minimize/maximize the difference of saliency between the edge pixel and each neighbor whether they both belong the salient object or not.
On ECSSD and DUT-OMROM benchmark datasets it lead to roughly 2-3% of increased precision at the given recall.
I hope it is clear enough!
